### ISSUE TITLE
Add virtual packages in run into the `hash`

### ISSUE THREAD

**Issue by wolfv** [prefix-dev/rattler-build#1073]
That is a relatively recent thing in conda-build.

**Comment by wolfv** [comment]
Discussion here: https://github.com/conda/conda-build/pull/4606

The idea is that this enables building platform specific noarch packages.

**Comment by wolfv** [comment]
The way this should be done in our case is that we add the `__osx: <matchspec>` or other virtual package specifiers to the `variant` table and then it will be automatically part of the hash. That is also simpler for the implementation.
We can also make it specific to only happen when building `noarch` packages.

**Comment by munechika-koyo** [comment]
I reached this issue when I used a platform-specific noarch package [`saxon-he`](https://github.com/conda-forge/saxon-he-feedstock?tab=readme-ov-file).
I would appreciate it if `rattler-build` could also handle this feature.
In the meantime, I will manually manage the build hash in the recipe.

**Comment by wolfv** [comment]
I already have some parts of the implementation in https://github.com/prefix-dev/rattler-build/pull/1122 - I will add a test to make sure it works!

**Comment by munechika-koyo** [comment]
Thank you for your prompt reply, @wolfv!
I'm excited for its release!

**Comment by wolfv** [comment]
This was fixed in #1122

**Comment by wolfv** [comment]
Not necessary to set anything wrt to `version_independent`. 

Both `noarch` should have this behavior (generic or python).

**Comment by munechika-koyo** [comment]
I tried to apply it to `saxon-he` feedstock, but it seems that the build hash did not change
```bash
> rattler-build build --recipe-dir ./recipe --output-dir=./output

 ╭─ Finding outputs from recipe
 │ Found 1 variants
 │ 
 │ Build variant: saxon-he-12.5-h5600cae_3
 │ 
 │ ╭─────────────────┬──────────╮
 │ │ Variant         ┆ Version  │
 │ ╞═════════════════╪══════════╡
 │ │ __unix          ┆ "__unix" │
 │ │ target_platform ┆ "noarch" │
 │ ╰─────────────────┴──────────╯
 │
 ╰─────────────────── (took 0 seconds)
```

I used the `rattler-build 0.41.0`

**Comment by wolfv** [comment]
I can see that the variant contains `__unix: __unix` which should have an effect on the build hash (`h5600cae`).

Is that not true?

**Comment by wolfv** [comment]
Yes, any virtual package in the run constraints should be part of the hash :)

**Comment by munechika-koyo** [comment]
I confirmed it when I tried `osx` constraints:
```bash
 ╭─ Finding outputs from recipe
 │ Found 1 variants
 │ 
 │ Build variant: saxon-he-12.5-hd9d1236_3
 │ 
 │ ╭─────────────────┬──────────╮
 │ │ Variant         ┆ Version  │
 │ ╞═════════════════╪══════════╡
 │ │ __osx           ┆ "__osx"  │
 │ │ target_platform ┆ "noarch" │
 │ ╰─────────────────┴──────────╯
 │
 ╰─────────────────── (took 0 seconds)
```
Thank you for implementing it.
What about documenting its features?
